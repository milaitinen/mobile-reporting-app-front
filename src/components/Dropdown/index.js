import Dropdown from './Dropdown';
import NestedDropDown from './NestedDropdown';

export { Dropdown, NestedDropDown };