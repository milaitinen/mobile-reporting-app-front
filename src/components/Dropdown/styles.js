import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({

    $activeBlue: '$active',
    $gray1: '$disabledPlaceholder',
    $gray2: '$placeholder',

    dropdownContainer: {
        margin: 5,
        flexDirection: 'row',
    },

    lowerDropdownStyleClass: {
        flex: 1,
        paddingTop: 8,
        paddingBottom: 8,
        paddingRight: 8,
        paddingLeft: 8,
    },

    dropdownButton: {
        width: 180,
        marginBottom: 10,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: '$dropdownRadius',
        borderWidth: '$containerBorderWidth',
        borderColor: '$inactive',
        backgroundColor: '$primaryWhite',
    },

    buttonContent: {
        width: 180,
        flexDirection: 'row',
        paddingHorizontal: 5,
        justifyContent: 'space-between',
        alignItems: 'center',
    },

    dropdownText: {
        marginVertical: 10,
        marginHorizontal: 6,
        fontSize: 16,
        fontFamily: '$primaryFont',
        color: '$placeholder',
        textAlign: 'center',
        textAlignVertical: 'center',
    },

    dropdownRow: {
        paddingHorizontal: 5,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        height: 40,
    },

    dropdownSeparator: {
        height: 1,
        backgroundColor: '$inactive',
    },

    dropdownRowText: {
        marginHorizontal: 4,
        fontSize: 16,
        fontFamily: '$primaryFont',
        color: '$placeholder',
        textAlignVertical: 'center',
    },

    dropStyleClass: {
        width: 180,
        height: 300,
        borderWidth: '$containerBorderWidth',
        borderColor: '$inactive',
        borderRadius: '$dropdownRadius',
    },

    icon: {
        alignSelf: 'flex-end',
    },

    disabled: {
        backgroundColor: '$disabled',
        borderColor: '$disabledBorder',
        borderWidth: 2,
    },

    disabledText: {
        color: '$disabledPlaceholder',
        fontFamily: '$primaryFont',
    }
});

export default styles;