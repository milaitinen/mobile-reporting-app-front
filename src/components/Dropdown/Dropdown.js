import React, { Component } from 'react';
import styles from './styles';
import ModalDropdown from 'react-native-modal-dropdown';
import { View, Text, TouchableHighlight } from 'react-native';
import { Icon } from 'react-native-elements';
import color from 'color';


class Dropdown extends Component {

    constructor(props) {
        super(props);
        this.state = {
            value: this.props.defaultValue
        };
    }

    renderButtonText = (rowData) => {
        return rowData;
    };

    renderRow = (rowData, rowID, highlighted) => {
        const evenRow = rowID % 2;
        return (
            <TouchableHighlight underlayColor={evenRow ? color('#DFF1F6').darken(0.2) : color('#87d8f6').darken(0.1)}>
                <View style={[styles.dropdownRow, { backgroundColor: evenRow ? '#DFF1F6' : 'white' }]}>
                    <Text style={[styles.dropdownRowText, highlighted && { color: '#8cc9e5' }]}>
                        {rowData}
                    </Text>
                    {this.state.value === rowData ? (
                        <Icon name={'check'} type={'feather'} color={'#8cc9e5'}/>
                    ) : (
                        null
                    )}
                </View>
            </TouchableHighlight>
        );
    };

    renderSeparator(sectionID, rowID) {
        if (rowID === this.props.options.length - 1) return;
        const key = 'spr_${rowID}';
        return (
            <View style={styles.dropdownSeparator} key={key}/>
        );
    }

    onSelect(idx, value) {
        this.setState({ value: value });
    }

    render() {
        return (
            <ModalDropdown
                ref={ ModalDrop => this.modalDropdown = ModalDrop }
                onPress={() => this.modalDropdown.show()}
                style={styles.dropdownButton}
                textStyle={styles.dropdownText}
                dropdownStyle={ styles.dropStyleClass }
                defaultValue={this.state.value}
                disabled={this.props.disabled}
                options={this.props.options}
                onSelect={(idx, value) => this.onSelect(idx, value)}
                renderButtonText={(rowData) => this.renderButtonText(rowData)}
                renderRow={this.renderRow.bind(this)}
                renderSeparator={(sectionID, rowID) => this.renderSeparator(sectionID, rowID)}
            >
                <View style={styles.buttonContent}>
                    <Text style={styles.dropdownText}>
                        {this.state.value}
                    </Text>
                    <Icon name={'expand-more'} color={'#adadad'} style={styles.icon}/>
                </View>
            </ModalDropdown>
        );
    }
}


export default Dropdown;