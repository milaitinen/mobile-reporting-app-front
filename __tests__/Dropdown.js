import 'react-native';
import React from 'react';
import { wrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Dropdown from '../src/components/Dropdown/Dropdown';

configure({ adapter: new Adapter() });

describe('Dropdown', () => {

    const wrapper = renderer.create(<Dropdown disabled={false} defaultValue={"some value"} options={['test1', 'test2']} />);
    const inst = wrapper.getInstance();

    it('renders correctly', () => {
        const tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("onSelect(0, 'test1')", () => {
        it('should set state value to test1 when called', () => {
            inst.onSelect(0, 'test1');
            expect(inst.state.value).toBe('test1');
        });
    });

});





