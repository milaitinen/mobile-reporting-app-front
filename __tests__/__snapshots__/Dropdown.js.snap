// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly 1`] = `
<View
  animated={true}
  defaultIndex={-1}
  defaultValue="some value"
  disabled={false}
  dropdownStyle={undefined}
  keyboardShouldPersistTaps="never"
  onPress={[Function]}
  onSelect={[Function]}
  options={
    Array [
      "test1",
      "test2",
    ]
  }
  renderButtonText={[Function]}
  renderRow={[Function]}
  renderSeparator={[Function]}
  scrollEnabled={true}
  showsVerticalScrollIndicator={true}
  style={undefined}
  textStyle={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        some value
      </Text>
      <View
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#adadad",
                "fontSize": 24,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          Óóè
        </Text>
      </View>
    </View>
  </View>
</View>
`;
